app/models/product.rb

```diff
@@ -1,4 +1,15 @@
 class Product < ApplicationRecord
   belongs_to :user
   validates_presence_of :name, :quantity, :expiry_date, :user_id
+
+  def list_expiring_products
+    User.all.each do |user|
+      products = list_expired_products(user.id, 7)
```

Comment:
That `7` looks a bit confusing there. You could pass things like `7.days.ago` and then use that timestamp to perform the actual query. It would look a bit more readable, in my opinion.